### PR TITLE
SshHandler introduced

### PR DESCRIPTION
**Pull request by xsmrcek**
SshHandler introduced.

### CODE REVIEW

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jbehavesupport.core.internal.ExampleTableConstraints.VERIFIER;
+import static org.jbehavesupport.core.internal.ExamplesTableUtil.convertTable;
+import static org.springframework.util.Assert.isTrue;
+import static org.springframework.util.Assert.notNull;
+
+@Slf4j
+public abstract class SshHandler {
```
Comment (paveljandejsek) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: Do we want/need abstract?

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java (hunk 1 above)
Comment (xsmrcek) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: No we don't :)

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+@Slf4j
+@RequiredArgsConstructor
+public class SshHandler {
+    private final TestContext testContext;
+    private final  ConfigurableListableBeanFactory beanFactory;
+    private final  ExamplesEvaluationTableConverter tableConverter;
+    private final  VerifierResolver verifierResolver;
+    private final  ContainsVerifier containsVerifier;
+    private final  NotContainsVerifier notContainsVerifier;
```
Comment (AmyKosinova) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: Decrease number of dependencies by pulling default verifiers into steps and present single entry point for `logContainsData` and `dataNotInLog`

Comment (xsmrcek) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: Done.

But I made dataNotInLog() method deprecated. As it is not needed in my point of view.

Comment (xsmrcek) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: If you agree i will remove it from documentation and tests.

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

File: jbehave-support-core/docs/Ssh.md [hunk 3]
```diff
@@ -21,6 +21,8 @@ If you have more specific needs not covered by this helper, feel free to combine
 Supported authentication methods are username with password or public key (optionally also with passphrase).
 In case that both password and public key are set then the public key gets precedence and password is ignored.
 
+Customization is allowed by overriding SshHandler and register custom handler as a bean.
```
Comment (paveljandejsek) [inline jbehave-support-core/docs/Ssh.md]:
```suggestion
Customization is allowed by overriding SshHandler and registering custom handler as a bean.
```

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+    public void saveLogStartTime(){
+        logStartTime = ZonedDateTime.now();
+    }
+
+    public void saveLogEndTime(){
+        logEndTime = ZonedDateTime.now().plusSeconds(1);
+    }
+
+    public void saveLogStartTimeOnSaved(ExpressionEvaluatingParameter<String> contextAlias) {
```
Comment (paveljandejsek) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: Minor but should probably be setLogStartTimeOnSaved?

```suggestion
    public void setLogStartTimeOnSaved(ExpressionEvaluatingParameter<String> contextAlias) {
```

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

File: jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+ * </pre>
+ */
+
+@Slf4j
+@RequiredArgsConstructor
+public class SshHandler {
+    private final TestContext testContext;
+    private final  ConfigurableListableBeanFactory beanFactory;
+    private final  ExamplesEvaluationTableConverter tableConverter;
+    private final  VerifierResolver verifierResolver;
```
Comment (AmyKosinova) [inline jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: double spaces

Comment (xsmrcek) [reply jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java]: no extra spaces anymore :)

**Comment by paveljandejsek** [comment]
Generated [reports](http://95.168.205.146:8080/reports/202/index.xml)

**Comment by paveljandejsek** [comment]
Successful build at 2019-11-13 10:46: https://travis-ci.org/EmbedITCZ/jbehave-support/builds/608085199?utm_medium=notification&utm_source=github_status

Travis for some reason did not report back to GH

**Pull request merged**